FurkannOzbek/application-hub
# [FEATURE] Implement Sort Functionality for Job Applications

# [FEATURE] Implement Sort Functionality for Applications

---

## Description

Added flexible sorting capabilities to the job applications dashboard, enabling users to organize their applications by different criteria.

Problem: Users needed to organise applications by different criteria
Impact: Enhanced organisation of job applications

---

## Changes Made

Frontend:
- DashboardPage: Added sort state management, and function of applying sorting after searching
- ApplicationsBoard: Updated to handle sorted applications
- Added utility function `sortApplications` 


---

## Screenshots (if applicable)

![screencapture-localhost-3000-user-2024-11-14-15_25_56](https://github.com/user-attachments/assets/26779bf7-c42c-4afa-9884-133aebce3f94)

![screencapture-localhost-3000-user-2024-11-14-15_18_06](https://github.com/user-attachments/assets/c131ea08-c11a-4933-8362-394708744a91)

---

## Testing Steps

1. Sort Functionality:
- Click sort dropdown and select different criteria:
- Test ascending/descending for each criteria
- Verify sort persistence between tab changes
